### ISSUE TITLE
DK64 doesn't work in LLE mode

### ISSUE THREAD

**Issue by xiartic** [project64/project64#2093]
Donkey Kong 64 hasn't worked properly since v2.4.0 came out when using LLE mode graphics. v2.3.2 is the last version where all worked fine. I've tested it with all gfx plugins that support LLE which are angrylion, parallel and gliden64. The issue is that the game runs too fast which leads to a crash. Same results when using the different RSP plugins such as cxd4, parallel-rsp as with Zilmar's.

**Comment by ghost** [comment]
Angryllion does not work with RARE games last I knew. Banjo Kazooie also has problems; however Diddy Kong Racing does not and works fine.

**Comment by TorutheRedFox** [comment]
try a different RSP plugin

**Comment by Squall-Leonhart** [comment]
narrow down the change this occurred in by testing 2.4 nightlies

https://www.pj64-emu.com/nightly-builds

**Comment by xiartic** [comment]
The earliest available v2.4.0 nightly "v2.4.0-1127-g185c658" there already has the issue unfortunately.

**Comment by xiartic** [comment]
Default plugins in both besides using parallel-rdp for graphics. v3.0.1 here crashes right after the logos.

v2.3.2.202:

https://user-images.githubusercontent.com/78692485/163708806-7a1b601e-0be0-4d8d-9328-023fd16d429e.mp4




v3.0.1-5664-2df3434:

https://user-images.githubusercontent.com/78692485/163708855-b7c59ddc-c6bd-4547-bf24-86ef3b14c246.mp4

**Comment by xiartic** [comment]
Went through other Rare titles as well and the results are following:

Blast Corps runs too fast.
Banjo Kazooie hangs on the start zoom-in.
Perfect Dark doesn't start.
Banjo-Tooie same as Banjo-Kazooie

**Comment by project64** [comment]
it might be due to RSP not running in parallel to the r4k so how the semaphore and RDP register handling might not be correct

**Comment by DerekTurtleRoe** [comment]
Have you tried Parallel-RSP with Parallel-RDP? If they work that way then I guess we know it's an RSP issue. Might also try cxd4's RSP plugin as well, since those seem to be the 2 most accurate RSP plugins.

**Comment by xiartic** [comment]
I said this in the issue message but yeah. I've tried Parallel-RSP with Parallel-RDP which gives the same broken result. 
RSP plugins seem to make no difference. 
Tested Zilmar's, CXD4's and Parallel-RSP with Angrylion, GLideN64 and Parallel-RDP which all have the issue when playing in LLE.

**Comment by DerekTurtleRoe** [comment]
> I said this in the issue message but yeah. I've tried Parallel-RSP with Parallel-RDP which gives the same broken result. RSP plugins seem to make no difference. Tested Zilmar's, CXD4's and Parallel-RSP with Angrylion, GLideN64 and Parallel-RDP which all have the issue when playing in LLE.

That's totally bizarre, I can't seem to replicate any of those issues you are experiencing. What system specs and OS version are you on?

**Comment by xiartic** [comment]
I'll list the details of my systems below:

1: OS: (Windows 10 21H1):

CPU: Intel Core i5-3350P

RAM: 8 GB

Graphics: Nvidia GTX 1060-3GB

2: OS: (Windows 11 22H2)

CPU: Intel Core i5-7200U

RAM: 8 GB

Graphics: Intel HD Graphics 620

**Comment by Squall-Leonhart** [comment]
I can replicate it with ease, the vi's show as running at 60 but its running way faster.

**Comment by topkex** [comment]
This game does not run in the latest nightly even with HLE graphics. Locks up after logo.

**Comment by Squall-Leonhart** [comment]
thats irrelevant to this lle issue and already has its own issue.
